dataframe/dataframe.go

```diff
@@ -656,6 +657,101 @@ func WithTypes(coltypes map[string]series.Type) LoadOption {
 	}
 }
 
+// Load creates a new DataFrame from arbitrary []structs
+func Load(i interface{}) DataFrame {
+	if i == nil {
+		return DataFrame{Err: fmt.Errorf("load: can't create DataFrame from <nil> value")}
+	}
+	tpy, val := reflect.TypeOf(i), reflect.ValueOf(i)
+	switch tpy.Kind() {
+	case reflect.Slice:
+		if tpy.Elem().Kind() != reflect.Struct {
+			return DataFrame{Err: fmt.Errorf("load: type %s (%s %s) is not supported, must be []struct", tpy.Name(), tpy.Elem().Kind(), tpy.Kind())}
+		}
+		if val.Len() == 0 {
+			return DataFrame{Err: fmt.Errorf("load: can't create DataFrame from empty slice")}
+		}
+		structs := []interface{}{}
+		for i := 0; i < val.Len(); i++ {
+			structs = append(structs, val.Index(i).Interface())
+		}
+		se := []series.Series{}
+		for i := range structs {
+			nse := seriesFromStruct(reflect.TypeOf(structs[i]), reflect.ValueOf(structs[i]))
```

Comment:
If I read this correctly this function is essentially creating a `[]series.Series` with only one element in each one of the returned `Series`. This in turn makes the access of the internal elements a bit more convoluted.

I would suggest that instead of wrapping and unwrapping `Series` and `Elements` you could transform the `[]Struct` to a `[]map[string]interface{}` that can then passed to `dataframe.LoadMaps` along with the given `LoadOptions`. This would greatly simplify this function, reduce the number of copies, remove the need of adding the helper functions `dataframe. seriesFromStruct` and `series.Elements` and make it easy to integrate with the rest of the library for future improvements.